ambry-api/src/main/java/com.github.ambry/commons/Threshold.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.commons;
+
+import com.github.ambry.utils.Pair;
+import java.util.Map;
+
+
+/**
+ * A data structure holding various thresholds based on different criteria.
+ */
+public class Threshold {
+  private final Map<PerformanceIndex, Pair<Long, Boolean>> perfIndexAndCriteria;
```

Comment:
I don't find the Pair construct very intuitive until I dive into it.  Suggest adding a simple class with fields thresholdValue and isUpperBound (or lower) to make it a bit clearer.  or a json map like Casey suggested.

Reply:
+1 Since this is passed in by callers to the constructor

Reply:
Thanks for both suggestions. Will use a separate `Criteria` class holding threshold value and `isUpperBound`.